File: python/y2019/day05.py
from aoc.core import solve, AocInput
from aoc.intcode import IntCodeMachine


def part1(input: AocInput) -> int:
    icm = IntCodeMachine.init(input.text)
    return IntCodeMachine.thermalEnvironmentSupervisionTerminal(icm, id=1)


def part2(input: AocInput) -> int:
    icm = IntCodeMachine.init(input.text)
    return IntCodeMachine.thermalEnvironmentSupervisionTerminal(icm, id=5)


def main() -> None:
    YEAR = "y2019"
    DAY = "Day05"
    print(f"Running year {YEAR.replace('y', '')} day {DAY.replace('Day', '')}")
    solve(YEAR, DAY, part1, expected=9961446, skipTest=True)
    solve(YEAR, DAY, part2, expected=742621, skipTest=True)

File: dotnet/2019/AdventOfCode.Y2019/Day05.cs
using AdventOfCode.Lib;

namespace AdventOfCode.Y2019;

[AocPuzzle(2019, 5, "Sunny with a Chance of Asteroids")]
public sealed class Day05 : IAocDay<int>
{
    public static int Part1(AocInput input) => input.Text
        .Pipe(IntCodeMachine.Init)
        .Pipe(icm => IntCodeMachine.ThermalEnvironmentSupervisionTerminal(icm, 1));

    public static int Part2(AocInput input) => input.Text
        .Pipe(IntCodeMachine.Init)
        .Pipe(icm => IntCodeMachine.ThermalEnvironmentSupervisionTerminal(icm, 5));
}